public/app/core/components/NativeScrollbar.tsx

```diff
@@ -2,28 +2,35 @@ import { css, cx } from '@emotion/css';
 import { useEffect, useRef } from 'react';
 
 import { config } from '@grafana/runtime';
-import { CustomScrollbar, useStyles2 } from '@grafana/ui';
+import { useStyles2 } from '@grafana/ui';
 
-type Props = Parameters<typeof CustomScrollbar>[0];
+export interface Props {
+  children: React.ReactNode;
+  onSetScrollRef?: (ref: ScrollRefElement) => void;
+  divId?: string;
+}
+
+export interface ScrollRefElement {
+  scrollTop: number;
+  scrollTo: (x: number, y: number) => void;
+}
 
 // Shim to provide API-compatibility for Page's scroll-related props
 // when bodyScrolling is enabled, this is a no-op
 // TODO remove this shim completely when bodyScrolling is enabled
```

Comment:
i guess this isn't true anymore 😅 

what do we envisage this component looking like once `bodyScrolling` is enabled by default and all toggle code removed? 🤔 does this basically become the abstraction you're talking about in the description? would something like `PageScrollHelper` make sense as name in the future?

(don't have to do any of this now, just trying to get it clear in my head where this component is heading)

Reply:
yea, my guess is we will need some kind of abstraction (component) and a context if we are to support switching between body scrolling and div scrolling.